lib/api/handlers/pokemon_handler.dart

```diff
@@ -121,17 +131,11 @@ class PokemonHandler {
     }
 
     if (response.statusCode == 200) {
-      final results = jsonDecode(response.body);
-      final List baseStatsResults = results['stats'];
-
-      return PokemonBaseStatsModel(
-        base_stat: baseStatsResults
-            .map((base_stat) => base_stat['base_stat'].toString())
-            .toList(),
-        name: baseStatsResults
-            .map((name) => name['stat']['name'].toString())
-            .toList(),
-      );
+      final List results = jsonDecode(response.body)['stats'];
+      final baseStatsResults =
+      results.map((stats) => PokemonBaseStatModel.fromJson(stats)).toList();
```

Comment:
then `final baseStatsResults = (results as List).map((stats) => PokemonBaseStatModel.fromJson(stats)).toList();`

Reply:
nvm, no need to store it in a variable. 

`return (results as List).map((stats) => PokemonBaseStatModel.fromJson(stats)).toList();`